sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -1356,6 +1358,18 @@ def test_asec():
     assert asec(x).rewrite(acsc) == -acsc(x) + pi/2
 
 
+def test_asec_is_real():
+    assert asec(S(1)/2).is_real is False
+    assert asec(S(1)).is_real is True
```

Comment:
The test with ``S(1)/2`` could be included, since there should be a difference to the old behaviour.